src/layer/tile/TileLayer.js

```diff
@@ -56,8 +56,8 @@ L.TileLayer = L.GridLayer.extend({
 	createTile: function (coords, done) {
 		var tile = document.createElement('img');
 
-		tile.onload = L.bind(this._tileOnLoad, this, done, tile);
-		tile.onerror = L.bind(this._tileOnError, this, done, tile);
+		L.DomEvent.on(tile, 'load', L.bind(this._tileOnLoad, this, done, tile));
+		L.DomEvent.on(tile, 'error', L.bind(this._tileOnError, this, done, tile));

```

Comment:
Firefox has real issues with `.onload`. It just doesn't fire during a `flyTo` animation, but somehow `L.DomEvent.on('load',...)` works flawlessly.